airbyte-integrations/connectors/source-relational-db/src/main/java/io/airbyte/integrations/source/relationaldb/InvalidCursorException.java

```diff
@@ -0,0 +1,26 @@
+package io.airbyte.integrations.source.relationaldb;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class InvalidCursorException extends RuntimeException {
+
+  public InvalidCursorException(final List<InvalidCursorInfo> tablesWithInvalidCursor) {
+    super("The following tables have invalid columns selected as cursor, please select a valid column as a cursor. " + tablesWithInvalidCursor.stream().map(InvalidCursorInfo::toString)
+        .collect(Collectors.joining(",")));
+  }
+
+  public record InvalidCursorInfo(String tableName, String cursorColumnName, String cursorSqlType) {
+
+    @Override
+    public String toString() {
```

Comment:
tiny nitpick: I'd prefer to define a new method `prettyString` instead of overriding toString (mostly in case someone actually wants to print one of these out for debugging or something)